### PR TITLE
Extraneous parts broken off from other DTA/TDA PRs

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
lgtm.

**Comment by jbrockmendel** [comment]
The failures here are all in openpyxl.  Is that causing problems in master right now?

**Comment by jreback** [comment]
yeah its not hitting master which is odd.

File: pandas/core/indexes/datetimes.py [hunk 1]
```diff
@@ -199,11 +198,15 @@ def _join_i8_wrapper(joinf, **kwargs):
...
 
-    tz = None
+    _tz = None
     _freq = None
     _comparables = ['name', 'freqstr', 'tz']
     _attributes = ['name', 'freq', 'tz']
 
+    # dummy attribute so that datetime.__eq__(DatetimeArray) defers
+    # by returning NotImplemented
+    timetuple = None
```
Comment (jorisvandenbossche) [inline pandas/core/indexes/datetimes.py]: This is not yet tested I suppose then?

Comment (jbrockmendel) [reply pandas/core/indexes/datetimes.py]: That is accurate.

**Pull request merged**